### PR TITLE
ESM and CommonJS release

### PR DESCRIPTION
**Pull request by sroussey**
This does much better compatibility between ESM and CommonJS modules in the JS ecosystem. I also found some functions not getting exported correctly. Not sure how tests ever passed. Also, I have the test be ESM and the bench be CJS just to make sure they stay working. The source is TS now though.

I'm going to have a look at usearch next. 

Please squash. :)

### CODE REVIEW

File: javascript/test.mjs [hunk 1]
```diff
@@ -1,9 +1,7 @@
 import test from 'node:test';
 import assert from 'node:assert';
-import build from 'node-gyp-build';
-import process from 'node:process';
 
-const simsimd = build(process.cwd());
+import * as simsimd from "./dist/esm/simsimd.js";
```
Comment (ashvardanian) [inline javascript/test.mjs:4]: Is it a good idea to hard-code relative paths like this?

File: javascript/simsimd.js
Comment (ashvardanian) [inline javascript/simsimd.js:1]: What if a person is running JavaScript runtime without TypeScript capabilities? Is it safe tor remove this file?

File: package.json [hunk 2]
```diff
@@ -24,13 +24,21 @@
   ],
   "scripts": {
     "install": "node-gyp-build",
-    "prebuild": "prebuildify --napi --strip --target=10.4.0",
+    "prebuild-single": "prebuildify --napi --strip --target=10.4.0",
```
Comment (ashvardanian) [inline package.json:27]: Which builds are we producing here? Are both 32-bit and 64-bit produced on x86 and Arm? Are they all needed?

File: javascript/test.mjs (hunk 1 above)
Comment (sroussey) [reply javascript/test.mjs:4]: Not if it were an example (which I am also thinking of adding), but for the test you want to test the generated code.

File: javascript/simsimd.js
Comment (sroussey) [reply javascript/simsimd.js:1]: Not even typescript people get the typescript file. The package.json deep links to both of the javascript versions (CJS and ESM) inside the dist folder that are generated from the typescript file. The typescript people get the whichever JS file and the .d.ts file. I wish TS had a config option to TS people would get the direct TS file but that is not a thing.

File: package.json (hunk 2 above)
Comment (sroussey) [reply package.json:27]: `prebuild-single` builds a single (normal) build
`prebuild-arm64` cross compiles (used for ubuntu)
`prebuild-darwin-x64+arm64` builds a universal binary for mac that includes both x86_64 and arm64

**Comment by sroussey** [comment]
Don't approve this just yet. I went to use this same logic on usearch and had issues. I think I may go to plain `prebuild` rather `prebuildify`.

**Pull request closed without merging**